app/client/src/sagas/PropertyPaneSagas.ts

```diff
@@ -0,0 +1,153 @@
+      propertyPath = "onFilesSelected";
+      propertyValue = `{{${currentAction.config.name}.run()}}`;
+      break;
+    case WidgetTypes.IFRAME_WIDGET:
+      propertyPath = "source";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.INPUT_WIDGET:
+      propertyPath = "defaultText";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.LIST_WIDGET:
+      propertyPath = "items";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.MAP_WIDGET:
+      propertyPath = "defaultMarkers";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.RADIO_GROUP_WIDGET:
+      propertyPath = "options";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.RATE_WIDGET:
+      propertyPath = "onRateChanged";
+      propertyValue = `{{${currentAction.config.name}.run()}}`;
+      break;
+    case WidgetTypes.RICH_TEXT_EDITOR_WIDGET:
+      propertyPath = "defaultText";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.DROP_DOWN_WIDGET:
+      propertyPath = "options";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.SWITCH_WIDGET:
+      propertyPath = "defaultSwitchState";
+      propertyValue = !!currentAction.data.body;
+      break;
+    case WidgetTypes.TABLE_WIDGET:
+      propertyPath = "tableData";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.TEXT_WIDGET:
+      propertyPath = "text";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    case WidgetTypes.VIDEO_WIDGET:
+      propertyPath = "url";
+      propertyValue = `{{${currentAction.config.name}.data}}`;
+      break;
+    default:
+      isValidProperty = false;
+      break;
+  }
+  if (queryId && isValidProperty) {
+    yield put(
+      updateWidgetPropertyRequest(
+        action.payload.widgetId,
+        propertyPath,
+        propertyValue,
+        RenderModes.CANVAS,
+      ),
+    );
+    yield put({
+      type: ReduxActionTypes.SHOW_PROPERTY_PANE,
+      payload: {
+        widgetId: action.payload.widgetId,
+        callForDragOrResize: undefined,
+        force: true,
+      },
+    });
+  } else {
+    queryId &&
+      Toaster.show({
+        text: "Binding on selection is not supported for this type of widget!",
+        variant: Variant.warning,
+      });
+  }
+  history.replace(BUILDER_PAGE_URL(applicationId, pageId, {}));
+}
+
+function* resetSnipingModeSaga() {
+  const currentURL = new URL(window.location.href);
+  const searchParams = currentURL.searchParams;
+  searchParams.delete("isSnipingMode");
+  searchParams.delete("bindTo");
+  history.replace({
+    pathname: currentURL.pathname,
+    search: searchParams.toString(),
+    hash: currentURL.hash,
+  });
```

Comment:
nitpick: this felt slightly better -
```suggestion
  history.replace({
    ...window.location,
    pathname: currentURL.pathname,
    search: searchParams.toString(),
  });
```